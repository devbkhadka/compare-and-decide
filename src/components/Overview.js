import React from 'react'
import { Typography, Box} from '@material-ui/core'
import {List, ListItem, ListItemText, ListItemIcon} from '@material-ui/core'
import {Star} from '@material-ui/icons'
import { FormattedMessage, defineMessages } from 'react-intl'


const steps = [
    defineMessages({step1: 'List all items from which you want to decide from with its attribute'}),
    defineMessages({step2: 'Select one prefered item based on intution'}),
    defineMessages({step3: `Now compare the current prefered item with each item one at a time. In each comparasion choose one item you prefer and eliminate other until all items are compared`}),
    defineMessages({step4:'Last prefered item remaining will be your choice. Hurray you have made the decision'})
]
const messages = defineMessages({
    title: "Make Decision in Three Easy Steps"
})

export default function Overview(props) {
    return <Box data-testid='overview'>
        <Typography variant='h4' style={{marginBottom:20}}>
            <FormattedMessage {...messages.title} />
        </Typography>
        
        <List>
            {steps.map((text, i)=>(<BulletPoint text={<FormattedMessage {...text[`step${i+1}`]} />} key={i}></BulletPoint>))}
        </List>
        
    </Box>
}


function BulletPoint({text}) {
    return <ListItem>
            <ListItemIcon>
                <Star />
            </ListItemIcon>
            <ListItemText
                primary={text}
            />
        </ListItem>
}