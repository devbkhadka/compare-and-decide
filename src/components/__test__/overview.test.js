import React from 'react'
import { render } from '@testing-library/react'

import Overview from '../overview'

describe("Test Overview Component", ()=>{
    it('Should render the content correctly', ()=> {
        const overview = render(<Overview></Overview>).baseElement

        expect(overview).toMatchSnapshot()
    })
}) 