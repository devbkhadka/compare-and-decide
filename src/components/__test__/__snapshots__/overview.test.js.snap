// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Overview Component Should render the content correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
        style="margin-bottom: 20px;"
      >
        Make Decision in Three Easy Steps
      </h4>
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              List all items from which you want to decide from with its attribute
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Select one prefered item based on intution
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Now compare the current prefered item with each item one at a time.In each comparasion choose one item you prefer and eliminate other until all items are compared
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Last prefered item remaining will be your choice. Hurray you have made the decision
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</body>
`;
